base/sort.jl

```diff
@@ -245,6 +248,13 @@ abstract Algorithm
 immutable InsertionSortAlg <: Algorithm end
 immutable QuickSortAlg     <: Algorithm end
 immutable MergeSortAlg     <: Algorithm end
+immutable PartialQuickSort <: Algorithm
+    k::Int
+end
+
+# partially sort until the  end of the range
+PartialQuickSort(r::OrdinalRange) = PartialQuickSort(last(r))
```

Comment:
That kinda happens on 443 inside `selectperm!`. There I say sort until the end of the range and then only return sorted indices within the range (`return ix[k]`)

It would probably be better to support this in the sort algorithm directly, but I didn't see an obvious way to extend @carlobaldassi code to handle it there.